packages/components/src/components/assets/CoinLogo/CoinLogo.tsx

```diff
@@ -1,38 +1,130 @@
+                setIsAccessible(false);
+            }
+        };
+
+        if (url) {
+            checkAccessibility();
+        }
+    }, [url]);
+
+    return isAccessible;
+};
+
+export const useAssetUrl = (coingeckoId: string, contractAddress?: string) => {
+    const fileName = contractAddress ? `${coingeckoId}_${contractAddress}` : coingeckoId;
+    const assetUrl = fileName ? `${UPDATED_ICONS_LIST_URL_BASE}${fileName}@2x.webp` : '';
+    const isAssetUrlAccessible = useCheckUrlAccessibility(assetUrl);
+    const iconUrl = fileName && isAssetUrlAccessible ? assetUrl : undefined;
+
+    return iconUrl;
+};
+
 export interface CoinLogoProps extends ImgHTMLAttributes<HTMLImageElement> {
-    symbol: CoinType;
+    symbol?: CoinType;
     className?: string;
     size?: number;
     index?: number;
+    coingeckoId?: string;
+    contractAddress?: string;
+    quality?: Quality;
 }
 
-const SvgWrapper = styled.div<Omit<CoinLogoProps, 'symbol'>>`
+const Wrapper = styled.div<{ $size: number }>`
     display: inline-block;
-    width: ${({ size }) => size}px;
-    height: ${({ size }) => size}px;
+    width: ${({ $size }) => $size}px;
+    height: ${({ $size }) => $size}px;
+    margin: ${spacingsPx.xxs};
 
     div {
-        width: ${({ size }) => size}px;
-        height: ${({ size }) => size}px;
-        line-height: ${({ size }) => size}px;
+        width: ${({ $size }) => $size}px;
+        height: ${({ $size }) => $size}px;
+        line-height: ${({ $size }) => $size}px;
     }
 `;
 
-export const CoinLogo = ({ symbol, className, size = 32, ...rest }: CoinLogoProps) => (
-    <SvgWrapper className={className} size={size} {...rest}>
-        <ReactSVG
-            src={COINS[symbol]}
-            beforeInjection={svg => {
-                svg.setAttribute('width', `${size}px`);
-                svg.setAttribute('height', `${size}px`);
-            }}
-            loading={() => <span className="loading" />}
-        />
-    </SvgWrapper>
+const StyledIconImage = styled.img<{ $size: number }>(
+    ({ $size }) => `
+        width: ${$size}px;
+        height: ${$size}px;
+    `,
+);
+
+type SVGCoinLogoProps = {
+    $symbol: CoinType;
+    $size?: number;
+};
+
+const SVGCoinLogo = ({ $symbol, $size = 32 }: SVGCoinLogoProps) => (
+    <ReactSVG
+        src={$symbol}
+        beforeInjection={svg => {
+            svg.setAttribute('width', `${$size}px`);
+            svg.setAttribute('height', `${$size}px`);
+        }}
+    />
 );
+
+export const CoinLogo = ({
+    symbol,
+    className,
+    size = 32,
+    coingeckoId = '',
+    contractAddress,
+    quality = 'medium',
+    ...rest
+}: CoinLogoProps) => {
+    const [isLoading, setIsLoading] = useState(true);
+
+    const iconUrl = useAssetUrl(coingeckoId, contractAddress);
+    const firstCharacter = coingeckoId && coingeckoId[0].toUpperCase();
+    const coinLogoSize = !symbol ? QUALITY_SIZE[quality] : size;
```

Comment:
Why is this duplicated here as well. I is both in [outer component](https://github.com/trezor/trezor-suite/pull/13416/files#diff-ff9cb18bace51536407e61224e5497845491a828f4287cc20dc61f02d2d4a947R204) and here.

Reply:
OK, I got that part. :+1: we need to use them independently.

However in the `AssetShareIndicator` the (https://github.com/trezor/trezor-suite/pull/13416/files#diff-ff9cb18bace51536407e61224e5497845491a828f4287cc20dc61f02d2d4a947R210) the `coinLogoSize` is passed into `size` and here it is again run thought same logic. 

At least I would pass just `size` and lets use `coinLogoSize` only for the `Container` in the `AssetShareIndicator`

Reply:
We can do that but here are a few why-s, why I didn't do that:
- the code of `AssetShareIndicator` would IMHO look kinda weird as one would ask why we pass `size` to one sub-component while `coinLogoSize` to the other one, for the same prop => inconsistency
- we still need to set the final `coinLogoSize` (or newly `assetLogoSize`) for `ProgressCircleIcon` anyway, cannot avoid that